uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/b2c/FacebookLoginComponentHandler.java

```diff
@@ -0,0 +1,101 @@
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.interaction.b2c;
+
+import android.widget.Button;
+
+import androidx.annotation.NonNull;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
+
+import com.microsoft.identity.client.ui.automation.interaction.ILoginComponentHandler;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+
+import org.junit.Assert;
+
+/**
+ * A login component handler for Facebook IdP
+ */
+public class FacebookLoginComponentHandler implements ILoginComponentHandler {
+
+    @Override
+    public void handleEmailField(@NonNull final String username) {
+        UiAutomatorUtils.handleInput("m_login_email", username);
+    }
+
+    @Override
+    public void handlePasswordField(@NonNull final String password) {
+        UiAutomatorUtils.handleInput("m_login_password", password);
+        handleNextButton();
+    }
+
+    @Override
+    public void handleBackButton() {
+        UiAutomatorUtils.pressBack();
+    }
+
+    @Override
+    public void handleNextButton() {
+        final UiDevice device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+        final UiObject nextBtn = device.findObject(
+                new UiSelector().className(Button.class).text("Log In")
+        );
+
+        try {
+            nextBtn.click();
+        } catch (UiObjectNotFoundException e) {
+            Assert.fail(e.getMessage());
+        }
+    }
+
+    @Override
+    public void handleAccountPicker(@NonNull final String username) {
+        throw new UnsupportedOperationException("Not supported for B2C Local Provider");
+    }
+
+    private UiObject getConsentScreen() {
+        return UiAutomatorUtils.obtainUiObjectWithResourceId("consentHeader");
+    }
+
+    @Override
+    public void confirmConsentPageReceived() {
+        throw new UnsupportedOperationException("Not supported for B2C Local Provider");
+    }
+
+    @Override
+    public void acceptConsent() {
+        throw new UnsupportedOperationException("Not supported for B2C Local Provider");
+    }
+
+    public void declineConsent() {
+        throw new UnsupportedOperationException("Not supported for B2C Local Provider");
+    }
+
+    @Override
+    public void handleSpeedBump() {
+        throw new UnsupportedOperationException("Not supported for B2C Local Provider");
+    }
+}
```

Comment:
These messages need to say Facebook instead of Local

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/939/commits/8d08d2646d79e6e2d044a047b4ec44b2ba9f8f6c